pkg/util/system/version_darwin.go

```diff
@@ -0,0 +1,45 @@
+// +build darwin
+
+package system
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// GetOsVersion returns a human friendly string of the current OS
+// in the case of an error this still returns a valid string for the details that can be found.
+func GetOsVersion() (string,  error) {
+	retVal := "unknown OSX version"
+
+	// you can not run the command once to get all this :-(
+	pn, err := runCommand("sw_vers", "-productName")
+	if err != nil {
+		return retVal, err
+	}
+	pv, err := runCommand("sw_vers", "-productVersion")
+	if err != nil {
+		retVal = fmt.Sprintf("%s unknown version", pn)
+	} else {
+		retVal = fmt.Sprintf("%s %s", pn, pv)
+	}
+
+	pb, err := runCommand("sw_vers", "-buildVersion")
+	if err != nil {
+		return fmt.Sprintf("%s unknown build", retVal), err
+	}
+	return fmt.Sprintf("%s build %s", retVal, pb), nil
+}
+
+
+func runCommand(command string, args ...string) (string, error) {
```

Comment:
This is duplicated code - assume you've added both because of the conditional building, which means not dup'd in runtime, but still in src control.
There is already a very similar function in cmd.CommonOptions.RunCommand (though I think the way it interacts with the verbose flag is not ideal). I actually starting refactoring some of this into the util package (just util.RunCommandBackground). How about adding your version (i.e. one that cares about the output) as util.RunCommand()?

Reply:
So having thought about this some more, I can potentially see cases where we may want to get the OS version from the `util` package and that would cause circular dependencies.
I could export this function in the system package and refactor all other usage from it but that seems a bit too much work for a this small piece of code.